File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/agents/interfaces/IBrowserService.java
package com.enterpriseagentos.backendjava.domain.features.agents.interfaces;

import java.util.UUID;
import java.util.concurrent.CompletableFuture;

import com.enterpriseagentos.backendjava.domain.features.agents.dtos.BrowserSessionState;

public interface IBrowserService {
    CompletableFuture<BrowserSessionState> getOrCreateAsync(UUID agentId);
    CompletableFuture<BrowserSessionState> getStateAsync(UUID agentId);
    CompletableFuture<BrowserSessionState> restartAsync(UUID agentId);
    CompletableFuture<Void> stopAsync(UUID agentId);
    CompletableFuture<String> getViewUrlAsync(UUID agentId);
}

File: apps/backend/src/Features/AgentHarness/Application/IBrowserService.cs
using OffceOs.Domain.Features.Browser;

namespace OffceOs.Application.Features.AgentHarness;

public interface IBrowserService
{
    Task<BrowserSessionState> GetOrCreateAsync(Guid agentId, CancellationToken ct = default);
    Task<BrowserSessionState?> GetStateAsync(Guid agentId, CancellationToken ct = default);
    Task<BrowserSessionState> RestartAsync(Guid agentId, CancellationToken ct = default);
    Task StopAsync(Guid agentId, CancellationToken ct = default);
    Task<string?> GetViewUrlAsync(Guid agentId, CancellationToken ct = default);
}
